src/client/components/equalizer/equalizer.js

```diff
@@ -0,0 +1,200 @@
+    } else {
+      soundFromFile.pause();
+    }
+    this.props.playPauseSoundFromFile();
+  }
+
+  startMuteStream = () => {
+    let {
+      audioStream,
+      audioContext,
+      analyser,
+      streamSource,
+      startMuteState
+    } = this.props.audioData;
+
+    if (!startMuteState) {
+      streamSource.connect(analyser);
+      analyser.connect(audioContext.destination);
+      //play/pause function doesn't work 
+      audioStream.play();
+      this.equaliserRun();
+      this.props.startMuteStreamAudio();
+    } else {
+      streamSource.disconnect(analyser);
+      audioStream.pause();
+      this.props.startMuteStreamAudio();
+    }
+  }
+
+  equaliserRun = () => {
+    requestAnimationFrame(this.renderEqualizer);
+  }
+
+  renderEqualizer = () => {
+    let {analyser, heightArray, ctx, numPoints} = this.state;
+    let flagColor = true;
+    analyser.getByteFrequencyData(heightArray);
+    let width = ctx.canvas.width;
+    let height = ctx.canvas.height;
+    let countcolumns = Math.floor(ctx.canvas.width / 52);
+    let columnwidth = Math.floor(5 / 6 * ctx.canvas.width / 52);
+    ctx.clearRect(0, 0, width, height);
+    for (let x = 0; x < width; x += countcolumns) {
+      let ndx = x * numPoints / width | 0;
+      let vol = heightArray[ndx];
+      let y = vol * height / 512;
+      this.roundedRect(ctx, x, height / 2, columnwidth, y, 2, flagColor);
+      flagColor = !flagColor;
+    }
+    requestAnimationFrame(this.renderEqualizer);
+  }
+
+  roundedRect = (ctx, x, y, width, height, radius, flagColor) => {
+    ctx.beginPath();
+    ctx.moveTo(x, y - height);
+    ctx.lineTo(x, y + height - radius);
+    ctx.arcTo(x, y + height, x + radius, y + height, radius);
+    ctx.lineTo(x + width - radius, y + height);
+    ctx.arcTo(x + width, y + height, x + width, y + height - radius, radius);
+    ctx.lineTo(x + width, y - height + radius);
+    ctx.arcTo(x + width, y - height, x + width - radius, y - height, radius);
+    ctx.lineTo(x + radius, y - height);
+    ctx.arcTo(x, y - height, x, y - height + radius, radius);
+    ctx.fillStyle = flagColor ? '#1ecea8' : '#93969f';
+    ctx.fill();
+  }
+  
+  uploadSoundInfoFromFile = (e) => {
+    let [file] = e.target.files;
+    let {audioContext: context, analyser} = this.props.audioData;
+    let audio = new Audio(URL.createObjectURL(file));
+    audio.loop = true;
+    audio.crossOrigin = "anonymous";
+    audio.addEventListener('canplay', () => {
+      try {
+        let source = context.createMediaElementSource(audio);
+        this.props.createAudioData({
+          audio,
+          file,
+          source,
+          name: file.name,
+          size: file.size,
+          type: file.type
+        });
+        source.connect(analyser);
+        analyser.connect(context.destination);
+      } catch (e) {
+        throw new Error(e);
+      }
+    });
+    audio.addEventListener('error', (e) => {
+      throw new Error(e);
+    });
+  }
+
+  render = () => ( 
```

Comment:
please fix formatting issue in this function 
for eg.
`<Graphicequaliser width = {this.props.audioData.widthCanvas} height = {this.props.audioData.heightCanvas onChangeWidth = {this.widthMerge}/>` on one line 

or
```
 <Graphicequaliser 
      width = {this.props.audioData.widthCanvas} 
      height = {this.props.audioData.heightCanvas}
      onChangeWidth = {this.widthMerge}
/>
```

Reply:
done